hack/update-codegen.sh

```diff
@@ -52,35 +95,81 @@ set -x
     github.com/tektoncd/pipeline/pkg/apis \
     pipeline:v1alpha1 \
     --go-header-file "${PROJECT_ROOT}/hack/boilerplate.go.txt" \
-    --output-base "${GEN_DIR}"
+    --output-base "${GEN_DIR}" \
+    || die "Code generation failed"
 set +x
 
-echo
-echo "## Move generated files ###########################"
-set -x
-mv "${GEN_DIR}/github.com/SAP/stewardci-core/pkg/client" "${PROJECT_ROOT}/pkg/"
-mv "${GEN_DIR}/github.com/SAP/stewardci-core/pkg/tektonclient" "${PROJECT_ROOT}/pkg/"
-cp -r "${GEN_DIR}/github.com/SAP/stewardci-core/pkg/apis" "${PROJECT_ROOT}/pkg/"
-rm -rf "${GEN_DIR}/github.com"
-set +x
+if [ "$VERIFY" = true ]; then
+    echo
+    echo "## Verifying generated sources ####################"
+    set -x
+    diff -Naupr ${GEN_DIR}/github.com/SAP/stewardci-core/pkg/client/ ${PROJECT_ROOT}/pkg/client/ || die "Regeneration required for clients"
+    diff -Naupr ${GEN_DIR}/github.com/SAP/stewardci-core/pkg/tektonclient/ ${PROJECT_ROOT}/pkg/tektonclient/ || die "Regeneration required for tektonclients"
+    diff -Naupr ${GEN_DIR}/github.com/SAP/stewardci-core/pkg/apis/steward/v1alpha1/zz_generated.deepcopy.go ${PROJECT_ROOT}/pkg/apis/steward/v1alpha1/zz_generated.deepcopy.go || die "Regeneration required for apis"
+    set +x
+else
+    echo
+    echo "## Move generated files ###########################"
+    set -x
+    mv "${GEN_DIR}/github.com/SAP/stewardci-core/pkg/client" "${PROJECT_ROOT}/pkg/" || die "Moving generated clients failed"
+    mv "${GEN_DIR}/github.com/SAP/stewardci-core/pkg/tektonclient" "${PROJECT_ROOT}/pkg/" || die "Moving generated tektonclients failed"
+    cp -r "${GEN_DIR}/github.com/SAP/stewardci-core/pkg/apis" "${PROJECT_ROOT}/pkg/" || die "Copying generated apis failed"
+    rm -rf "${GEN_DIR}/github.com" || die "Cleanup gen dir failed"
+    set +x
+fi
+
 
 echo
-echo "## Generate mocks for package 'k8s' ###############"
-set -x
-"$GOPATH_1/bin/mockgen" \
-    -copyright_file="${PROJECT_ROOT}/hack/boilerplate.go.txt" \
-    -destination="${PROJECT_ROOT}/pkg/k8s/mocks/mocks.go" \
-    -package=mocks \
-    github.com/SAP/stewardci-core/pkg/k8s \
-    PipelineRun,ClientFactory,PipelineRunFetcher,NamespaceManager
-set +x
+if [ "$VERIFY" = true ]; then
```

Comment:
I would do the generation outside the if to avoid redundancy:

if verify
  gen_dir = temp
else 
  gen_dir = real
fi

codegeneration(gen_dir)

if verify
diff (temp,real)
fi

Reply:
see https://github.com/SAP/stewardci-core/pull/19/commits/037925244fcd1c755f475316067d8cae410780d1